src/util/OGCFilter.js

```diff
@@ -0,0 +1,328 @@
+                '</wfs:Query>' +
+            '</wfs:GetFeature>',
+
+
+        /**
+         * Given an array of ExtJS Filters, this method will return an OGC
+         * compliant filter which can be used for WMS requests
+         * @param {array} filters array containing all `Ext.grid.filters.filter`
+         *   that should be converted
+         * @param {string} combinator The combinator used for combining multiple
+         *   filters. Can be 'and' or 'or'
+         * @return {string} The OGC Filter XML
+         */
+        getOGCWMSFilterFromExtJSFilter: function(filters, combinator) {
+            return GeoExt.util.OGCFilter.getOGCFilterFromExtJSFilter(
+                filters, 'wms', combinator);
+        },
+
+        /**
+         * Given an array of ExtJS Filters, this method will return an OGC
+         * compliant filter which can be used for WFS requests
+         * @param {array} filters array containing all `Ext.grid.filters.filter`
+         *   that should be converted
+         * @param {string} combinator The combinator used for combining multiple
+         *   filters. Can be 'and' or 'or'
+         * @param {string} wfsVersion The WFS version to use, either `1.0.0`,
+         *   `1.1.0` or `2.0.0`
+         * @return {string} The OGC Filter XML
+         */
+        getOGCWFSFilterFromExtJSFilter: function(filters, combinator,
+            wfsVersion) {
+            return GeoExt.util.OGCFilter.getOGCFilterFromExtJSFilter(
+                filters, 'wfs', combinator, wfsVersion);
+        },
+
+        /**
+         * Given an ExtJS Filter, this method will return an OGC compliant
+         * filter which can be used for WMS or WFS queries
+         * @param {array} filters array containing all `Ext.grid.filters.filter`
+         *   that should be converted
+         * @param {string} type The OGC type we will be using, can be
+         *   `wms` or `wfs`
+         * @param {string} combinator The combinator used for combining multiple
+         *   filters. Can be 'and' or 'or'
+         * @param {string} wfsVersion The WFS version to use, either `1.0.0`,
+         *   `1.1.0` or `2.0.0`
+         * @return {string} the OGC Filter as XML tring
+         */
+        getOGCFilterFromExtJSFilter: function(filters, type, combinator,
+            wfsVersion) {
+            if (!Ext.isDefined(filters) || !Ext.isArray(filters)) {
+                Ext.Logger.error('Invalid filter argument given to ' +
+                  'GeoExt.util.OGCFilter. You need to pass an array of ' +
+                  '"Ext.grid.filters.filter"');
+                return;
+            }
+
+            var omitNamespaces = false;
+            // filters for WMS layers need to omit the namespaces
+            if (!Ext.isEmpty(type) && type.toLowerCase() === 'wms') {
+                omitNamespaces = true;
+            }
+            var ogcFilters = [];
+            Ext.each(filters, function(filter) {
```

Comment:
Filters is an array of Ext.util.Filter ([see](https://docs.sencha.com/extjs/6.2.1/classic/Ext.grid.Panel.html#event-filterchange)). This seems to have getter for the properties needed, see [here for getOperator](https://docs.sencha.com/extjs/6.2.1/classic/Ext.util.Filter.html#method-getOperator)

Reply:
right, fixed